src/WebJobs.Extensions.DurableTask/Listener/TaskActivityShim.cs

```diff
@@ -54,7 +54,26 @@ public override async Task<string> RunAsync(TaskContext context, string rawInput
                 functionType: FunctionType.Activity,
                 isReplay: false);
 
-            FunctionResult result = await this.executor.TryExecuteAsync(triggerInput, CancellationToken.None);
+            FunctionResult result;
+            try
+            {
+                result = await this.executor.TryExecuteAsync(triggerInput, CancellationToken.None);
+            }
+            catch (Exception e)
+            {
+                this.config.TraceHelper.FunctionFailed(
+                    this.config.Options.HubName,
+                    this.activityName,
+                    instanceId,
+                    $"An internal error occurred while attempting to execute this function. The execution will be aborted and retried. Details: {e}",
+                    functionType: FunctionType.Activity,
+                    isReplay: false);
+
+                // This will abort the execution and cause the message to go back onto the queue for re-processing
+                throw new SessionAbortedException(
```

Comment:
Is this the only change unique to this PR?

Reply:
This commit is the only change https://github.com/Azure/azure-functions-durable-extension/pull/1139/commits/a2d5460a01951c36bc4ae3c37e5ccfe0d0893aff